cmd/ipfs/daemon.go

```diff
@@ -488,3 +503,16 @@ func merge(cs ...<-chan error) <-chan error {
 	}()
 	return out
 }
+
+func runGC(req cmds.Request, repo repo.Repo, node *core.IpfsNode) error {
+	disableGC, _, err := req.Option(disableGCKwd).Bool()
+	if err != nil {
+		return err
+	}
+	if disableGC {
+		return nil
+	}
+
+	ctx := req.Context()
+	return fsrepo.PeriodicGC(ctx, repo, node)
```

Comment:
i think the `PeriodicGC` function should either:
```go
// be called on repo
myrepo.PeriodicGC(ctx, node)

// or be called on repo package
repo.PeriodicGC(ctx, myrepo, node)
```

basically, not fsrepo, as that's a repo implementation specific thing. It will get complicated with S3 datastore, etc.

I think we should either:
1. formally extend the `Repo` interface to have a `PeriodicGC` function
2. formally extend the `Repo` interface to have a `PeriodicTasks` function
3. make disabling periodic gc an _option_ on repo init, and hide the `PeriodicGC` function call within the repo constructor (this is likely to be easier on the user for some cases, but harder + magical + unpredictable in others).

probably 1 or 2 is best?